applications/ShapeOptimizationApplication/tests/test_ShapeOptimizationApplication.py

```diff
@@ -0,0 +1,90 @@
+# ==============================================================================
+# Imports
+# ==============================================================================
+
+# Making KratosMultiphysics backward compatible with python 2.6 and 2.7
+from __future__ import print_function, absolute_import, division
+
+# import Kratos
+import KratosMultiphysics
+import KratosMultiphysics.ShapeOptimizationApplication as ShapeOptimizationApplication
+
+# Import Kratos "wrapper" for unittests
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+# Import necessary external applications
+try:
+    import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+    import KratosMultiphysics.ExternalSolversApplication as ExternalSolversApplication
+    import KratosMultiphysics.EigenSolversApplication as EigenSolversApplication
```

Comment:
If eigen is not available the test will not be executed at all => the case for the nightly testing!